packages/vite/src/node/packages.ts

```diff
@@ -60,8 +61,8 @@ export function resolvePackageData(
   }
   let pkgPath: string | undefined
   try {
-    pkgPath = resolveFrom(`${id}/package.json`, basedir, preserveSymlinks)
-    pkg = loadPackageData(pkgPath, true, packageCache)
+    pkgPath = resolveFrom(`${id}/package.json`, basedir, true)
```

Comment:
Preserve symlinks when using `resolveFrom` so we have the symlink path inside `loadPackageData`.